examples/newsreaderExample/src/main/java/io/realm/examples/newsreader/model/network/NYTimesDataLoader.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.newsreader.model.network;
+
+import java.text.ParsePosition;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.examples.newsreader.model.DefaultTransactionCallback;
+import io.realm.examples.newsreader.model.entity.NYTimesStory;
+import retrofit.JacksonConverterFactory;
+import retrofit.Retrofit;
+import retrofit.RxJavaCallAdapterFactory;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.functions.Action1;
+import rx.schedulers.Schedulers;
+import rx.subjects.BehaviorSubject;
+import timber.log.Timber;
+
+
+/**
+ * Class that handles network requests for the New York Times API
+ */
+public class NYTimesDataLoader {
+
+    String[] nytSections = new String[]{
+            "home",
+            "world",
+            "national",
+            "politics",
+            "nyregion",
+            "business",
+            "opinion",
+            "technology",
+            "science",
+            "health",
+            "sports",
+            "arts",
+            "fashion",
+            "dining",
+            "travel",
+            "magazine",
+            "realestate"
+    };
+
+    private NYTimesService nyTimesService;
+    private SimpleDateFormat inputDateFormat = new SimpleDateFormat("yyyy-MM-d'T'HH:mm:ssZZZZZ");
+    private SimpleDateFormat outputDateFormat = new SimpleDateFormat("MM-dd-yyyy");
+    private String apiKey;
+    private Realm realm;
+    private BehaviorSubject<Boolean> networkInUse;
+
+    public NYTimesDataLoader() {
+        Retrofit retrofit = new Retrofit.Builder()
+                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
+                .addConverterFactory(JacksonConverterFactory.create())
+                .baseUrl("http://api.nytimes.com/")
+                .build();
+        nyTimesService = retrofit.create(NYTimesService.class);
+    }
+
+    public void loadAllData(final Realm realm, String apiKey, BehaviorSubject<Boolean> networkUsage) {
+        this.apiKey = apiKey;
+        this.realm = realm;
+        this.networkInUse = networkUsage;
+        loadNextSection(0);
```

Comment:
This was a copy/paste from the example that show-cased the API, but I agree, it is a bit complex. Maybe I should add a Spinner to the Actionbar that allowed you to switch between the different sections. That would also make for a more full-fledged example and could add a bit more to the Model.